src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionServiceTest.java

```diff
@@ -233,6 +240,21 @@ public void shouldThrowExceptionWithUnknownLevelOfAssurance() throws Exception {
         nonMatchingAssertionService.translateSuccessResponse(ImmutableList.of(authnAssertion, mdsAssertion), "requestId", LEVEL_2, "default-entity-id");
     }
 
+    @Test
+    public void shouldPerformHashing() {
+
+        Assertion authnAssertion = anAuthnStatementAssertion(IdaAuthnContext.LEVEL_2_AUTHN_CTX, "requestId").buildUnencrypted();
+        Assertion mdsAssertion = aMatchingDatasetAssertionWithSignature(emptyList(), anIdpSignature(), "requestId").buildUnencrypted();
+
+        final String issuerId = authnAssertion.getIssuer().getValue();
+        final String nameId = authnAssertion.getSubject().getNameID().getValue();
+
+        nonMatchingAssertionService.translateSuccessResponse(ImmutableList.of(authnAssertion, mdsAssertion), "requestId", LEVEL_2, "default-entity-id");
+
+        verify(userIdHashFactory, times(1)).hashId(issuerId,nameId, Optional.of(AuthnContext.LEVEL_2));
```

Comment:
As I've alluded to in another comment, rather than checking that the userIdHashFactory gets used, we should specify a return value for it and check that the response contains that value for the hash.